python-backend/app/api/users/minespace/resources/minespace_user_mine.py

```diff
@@ -0,0 +1,49 @@
+import uuid
+
+from flask_restplus import Resource, reqparse
+from flask import request
+from app.extensions import api
+from ....utils.access_decorators import requires_role_mine_admin
+from ....utils.resources_mixins import UserMixin, ErrorMixin
+from ..models.minespace_user import MinespaceUser
+from ..models.minespace_user_mine import MinespaceUserMine
+from app.extensions import db
+
+
+class MinespaceUserMineResource(Resource, UserMixin, ErrorMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument('mine_guid', type=str, required=True)
+
+    @api.doc(params={'user_id': 'User id.', 'mine_guid': 'MDS Mine Guid'})
+    @requires_role_mine_admin
+    def post(self, user_id=None, mine_guid=None):
+        if not user_id:
+            return self.create_error_payload(400, "user_id not found"), 400
+        if mine_guid:
+            return self.create_error_payload(400, "unexpected mine_guid"), 400
+        data = self.parser.parse_args()
+        guid = uuid.UUID(data.get('mine_guid'))  #ensure good formatting
+
+        user = MinespaceUser.find_by_id(user_id)
+        user.mines.append(MinespaceUserMine(user_id=user.id, mine_guid=guid))
+        user.save()
```

Comment:
I would expect we should wrap these save calls in a `try` to gracefully handle a failure

Reply:
done.